[IR][unittests] Replace of PointerType::get(Type) with opaque version (NFC)

junlarsen opened the pull request:
In accordance with https://github.com/llvm/llvm-project/issues/123569

junlarsen committed:
[IR][unittests] Replace of PointerType::get(Type) with opaque version (NFC)

In accordance with https://github.com/llvm/llvm-project/issues/123569

junlarsen committed:
[fix] run clang format

nikic commented on llvm/unittests/FuzzMutate/RandomIRBuilderTest.cpp:
```diff
@@ -328,7 +328,7 @@ TEST(RandomIRBuilderTest, createStackMemory) {
       DoubleTy,
       PointerType::get(Ctx, 0),
-      PointerType::get(Int32Ty, 0),
+      PointerType::get(Ctx, 0),
```
Would be better to drop the duplicate (and the corresponding line in Init).

nikic commented on llvm/unittests/IR/InstructionsTest.cpp:
```diff
@@ -406,8 +406,8 @@ TEST(InstructionsTest, CastCAPI) {
-  Type *Int8PtrTy = PointerType::get(Int8Ty, 0);
-  Type *Int32PtrTy = PointerType::get(Int32Ty, 0);
+  Type *Int8PtrTy = PointerType::get(C, 0);
+  Type *Int32PtrTy = PointerType::get(C, 0);
```
We should combine these into one PtrTy and then drop any tests that became redundant as a result.

junlarsen replied:
I also took care of some the same issue in the test above and below.

junlarsen committed:
[IR][unittests] Deduplicate pointer tests

A lot of the old tests test the same thing twice because pointers are opaque. This removes the deduplication

junlarsen committed:
[IR][unittests] Fix regressions from removal of duplicate pointer ty

junlarsen commented:
> Looks like a unit test is failing.

I accidentally used the wrong renamed variable when I was doing this last night. It should be good to go now.

nikic reviewed (approved):
LGTM
